src/components/modals/login_modal.tsx

```diff
@@ -99,19 +152,52 @@ const LoginModal: React.FC<LoginModalProps> = ({
 
 				<div className="flex justify-between mt-4">
 					<button
-						onClick={onClose}
+						onClick={() => {
+							resetLoginState(); // Reset state when modal closes
+							onClose(); // Close the modal
+						}}
 						className="text-sm text-gray-600 hover:underline"
 					>
 						Cancel
 					</button>
 					<button
 						onClick={handleLogin}
-						className="bg-sky-600 text-white px-4 py-2 rounded-lg hover:bg-sky-700 text-sm"
+						disabled={isLoggingIn}
+						className={`bg-sky-600 text-white px-4 py-2 rounded-lg text-sm ${
+							isLoggingIn ? "opacity-50 cursor-not-allowed" : "hover:bg-sky-700"
+						}`}
 					>
-						Login
+						{isLoggingIn ? "Logging in..." : "Login"}
+					</button>
+				</div>
+
+				<div className="flex justify-center mt-4">
+					<button
+						onClick={() => setShowResetModal(true)}
+						className="text-xs text-sky-600 hover:underline"
+					>
+						Forgot Password?
 					</button>
 				</div>
 			</div>
+
+			{/* Reset Password Modal */}
+			<ResetPasswordModalProps
```

Comment:
```suggestion
			<ResetPasswordModal
```
Remove props from component name